lib/manager/components/transform/NormalizeField.js

```diff
@@ -0,0 +1,227 @@
+// @flow
+
+import Icon from '@conveyal/woonerf/components/icon'
+import React, { Component } from 'react'
+import {
+  Button,
+  Checkbox,
+  ControlLabel,
+  FormGroup,
+  OverlayTrigger,
+  Table,
+  Tooltip
+} from 'react-bootstrap'
+import Select from 'react-select'
+
+import GtfsFieldSelector from '../GtfsFieldSelector'
+import SubstitutionRow from './SubstitutionRow'
+
+import type {
+  NormalizeFieldFields,
+  ReactSelectOption,
+  Substitution,
+  TransformProps
+} from '../../../types'
+
+/**
+ * Component that renders input fields for the NormalizeFieldTransformation.
+ */
+export default class NormalizeField extends Component<TransformProps<NormalizeFieldFields>, NormalizeFieldFields> {
+  componentDidMount () {
+    this._updateErrors()
+  }
+
+  _onChangeFieldToNormalize = (fieldOption: ReactSelectOption) => {
+    this._updateTransformation({fieldName: fieldOption.value})
+  }
+
+  _onChangeCapitalize = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    this._updateTransformation({capitalize: evt.target.checked})
+  }
+
+  _onChangeCapitalizationExceptions = (values: Array<{value: string}>) => {
+    this._updateTransformation({capitalizationExceptions: values.map(v => v.value)})
+  }
+
+  _onChangeSubstitution = (substitution: Substitution, index: number) => {
+    const newSubstitutions = [].concat(this.props.transformation.substitutions)
+    newSubstitutions[index] = substitution
+    this._updateTransformation({substitutions: newSubstitutions})
+  }
+
+  _onRemoveSubstitution = (index: number) => {
+    const newSubstitutions = [].concat(this.props.transformation.substitutions)
+    newSubstitutions.splice(index, 1)
+    this._updateTransformation({substitutions: newSubstitutions})
+  }
+
+  _onNewSubstitution = () => {
+    this._updateTransformation({substitutions: [...[].concat(this.props.transformation.substitutions), {
```

Comment:
Given a new transformation, when this is executed, then the `this.props.transformation.substitutions` is null which causes the array to become `[null]` which leads to a downstream error anywhere that properties on this supposed expected substitution object are checked. This results in the Add button not adding any substitutions and just error'ing out.

Reply:
Thanks for finding this bug! Fixed in. https://github.com/ibi-group/datatools-ui/pull/649/commits/a595d1a6b4f0f173dc330fb9987f0489e49d20ee.